js/lib/ui.js

```diff
@@ -209,6 +209,36 @@ elgg.ui.initHoverMenu = function(parent) {
 
 	// avatar contextual menu
 	$(".elgg-avatar > .elgg-icon-hover-menu").live('click', function(e) {
+		var $placeholder = $(this).parent().find(".elgg-menu-hover.elgg-ajax-loader");
+
+		if ($placeholder.length) {
+			// select all similar placeholders
+			var $all_placeholders = $(".elgg-menu-hover[rel='" + $placeholder.attr("rel") + "']");
+
+			// find the <ul> that contains data for this menu
+			var $ul = $all_placeholders.filter('[data-elgg-menu-data]');
+
+			if ($ul.length) {
+				elgg.get('ajax/view/navigation/menu/user_hover/contents', {
+					data: $ul.data('elggMenuData'),
+					success: function(data) {
```

Comment:
I think it's not that bad here just yet. IMHO will be more readable this way, though we might think of some specific approach by ie. making most non-trivial callbacks as separate variable